sklearn/manifold/_locally_linear.py

```diff
@@ -67,7 +67,7 @@ def barycenter_weights(X, Y, indices, reg=1e-3):
         else:
             R = reg
         G.flat[::n_neighbors + 1] += R
-        w = solve(G, v, sym_pos=True)
+        w = solve(G, v, sym_pos=True, check_finite=False)
```

Comment:
maybe we should check that the parameter `reg` is finite ?

Reply:
I would have initially said that this is up to the user/client responsibility but it's probably better to keep this check for user accessibility, yes.